radix-engine/src/transaction/transaction_receipt.rs

```diff
@@ -230,8 +226,16 @@ impl TransactionReceipt {
     }
 
     pub fn output<T: ScryptoDecode>(&self, nth: usize) -> T {
-        scrypto_decode::<T>(&self.expect_commit_success()[nth][..])
-            .expect("Wrong instruction output type!")
+        match &self.expect_commit_success()[nth] {
```

Comment:
is there no need to check if nth is not out of bounds?

Reply:
Don't think it's too necessary here as the caller should always know how many instructions/outputs there are. It may be that there is a better abstraction that would avoid this invariant being upheld by the client but don't think it's too important now.

Reply:
ok, sounds reasonable